### ISSUE TITLE
v0.9.30 on lollipop and marshmallow asks me to sign in twice

### ISSUE THREAD

**Issue by KhinRunite** [playgameservices/play-games-plugin-for-unity#986]
This behavior is confirmed found on lollipop and marshmallow devices. After hitting cancel on the first Auth UI, it will appear again. Hitting cancel on the second UI ends the authentication phase. I looked at the logs, and found two instances of this line:

I/Unity   (19565): there are pending auth callbacks - starting AuthUI

But I only called authenticate once:

```
        Social.localUser.Authenticate((bool success) => {
            // handle success or failure
            IsOnline = success;
            Debug.Log("authenticate return: " + success);

            if ( callback != null )
            {
                callback.Invoke( success );
            }
        });
```

The success/failure handler is also executed only ONCE, as you can see in the log.

The issue is not observed on Kitkat devices.

If you hit Ok instead of Cancel on the first sign in, you will be signed in (provided that you are an eligible tester account), and the second auth UI will not appear anymore.

My logfile is attached.
[logcat.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/133627/logcat.txt)

**Comment by Ploulack** [comment]
I have the same issue.

**Comment by KhinRunite** [comment]
It may be useful to know that if I don't enable saved games when initializing the plugin, the issue does not occur. I made sure that the the 'Saved games' toggle is turned ON in the developer console. I also have the Drive API added.

**Comment by claywilkinson** [comment]
When you are talking about OK/Cancel - are you referring to the consent screen? What version of play-services are you using, and which version android are you targeting?  

Are you adding any other permissions directly or via a plugin?

**Comment by NickBack** [comment]
It's happening on us too. To recreate, you have to use saved games, call authenticate once, and then when the Auth UI pops up, either cancel or tap outside the GUI so it cancel the authentication.

After a couple of seconds the UI pops up again.

In the logs we're also seeing "there are pending auth callbacks - starting AuthUI" twice.

We are building using api 22.

**Comment by claywilkinson** [comment]
This fixed in 0.9.37
